src/compile/legend/encode.ts

```diff
@@ -1,34 +1,25 @@
+import {SymbolEncodeEntry} from 'vega';
 import {isArray} from 'vega-util';
-
 import {Channel, COLOR, NonPositionScaleChannel, OPACITY, SHAPE} from '../../channel';
-import {
-  Conditional,
-  FieldDef,
-  FieldDefWithCondition,
-  hasConditionalValueDef,
-  isTimeFieldDef,
-  isValueDef,
-  MarkPropFieldDef,
-  ValueDef,
-  ValueDefWithCondition,
-} from '../../fielddef';
+import {Conditional, FieldDef, FieldDefWithCondition, hasConditionalValueDef, isTimeFieldDef, isValueDef, MarkPropFieldDef, ValueDef, ValueDefWithCondition} from '../../fielddef';
 import {AREA, BAR, CIRCLE, FILL_STROKE_CONFIG, GEOSHAPE, LINE, POINT, SQUARE, TEXT, TICK} from '../../mark';
 import {ScaleType} from '../../scale';
 import {keys} from '../../util';
-import {LegendType, VgEncodeEntry} from '../../vega.schema';
 import {applyMarkConfig, timeFormatExpression} from '../common';
 import * as mixins from '../mark/mixins';
 import {UnitModel} from '../unit';
+import {LegendComponent} from './component';
+
 
-export function symbols(fieldDef: FieldDef<string>, symbolsSpec: any, model: UnitModel, channel: Channel, type: LegendType): VgEncodeEntry {
-  if (type === 'gradient') {
+export function symbols(fieldDef: FieldDef<string>, symbolsSpec: any, model: UnitModel, channel: Channel, legendCmp: LegendComponent): SymbolEncodeEntry {
+  if (legendCmp.get('type') === 'gradient') {
     return undefined;
   }
 
   let out = {
     ...applyMarkConfig({}, model, FILL_STROKE_CONFIG),
     ...mixins.color(model)
-  };
+  } as SymbolEncodeEntry;
```

Comment:
Why do we have to cast here.  Can't we assign type to out? (Add comment if we can't)

Reply:
Added a fixme